cmd/cstor-pool-mgmt/pool/pool.go

```diff
@@ -107,10 +126,19 @@ func createPoolBuilder(cStorPool *apis.CStorPool) []string {
 
 	poolNameUID := string(PoolPrefix) + string(cStorPool.ObjectMeta.UID)
 	createAttr = append(createAttr, poolNameUID)
-	// To generate mirror disk0 disk1 mirror disk2 disk3 format.
-	for i, disk := range cStorPool.Spec.Disks.DiskList {
-		if cStorPool.Spec.PoolSpec.PoolType == "mirrored" && i%2 == 0 {
-			createAttr = append(createAttr, "mirror")
+	if poolType == "" || poolType == "striped" {
+		for _, disk := range diskList {
+			createAttr = append(createAttr, disk)
+		}
+		return createAttr
+	}
+	// To generate pool of the following types:
+	// mirrored (grouped by multiples of 2): mirror disk1 disk2 mirror disk3 disk4
+	// raidz (grouped by multiples of 3): raidz disk1 disk2 disk3 raidz disk 4 disk5 disk6
+	// raidz2 (grouped by multiples of 6): raidz2 disk1 disk2 disk3 disk4 disk5 disk6
+	for i, disk := range diskList {
+		if i%poolGroupByMap[poolType] == 0 {
```

Comment:
don't understand from poolGroupByMap.. use `groupSizeByPoolType`

Reply:
How about `poolGroupSizeMap` ?
My understanding is following:
Suffixing first class objects by their type is what we have seen as a general pattern.
e.g. 
`predicateFunc` --> Tells, a first class function object/type
`someInfoMap`  -->  Tells, a utility map
`someInfoList`  --> Tells, a list
and so on